libs/editor/editor_notes.cpp

```diff
@@ -181,12 +181,24 @@ void Notes::Upload(osm::OsmOAuth const & auth)
 
         LOG(LINFO, ("A note uploaded with id", id));
       }
+      catch (osm::OsmOAuth::NetworkError const & e)
+      {
+        ulock.lock();  // important: restore mutex before returning
+        LOG(LWARNING, ("Network error while uploading note:", e.what()));
+        return;
+      }
       catch (osm::ServerApi06::ServerApi06Exception const & e)
```

Comment:
RootException is the common parent for all our exceptions.
```suggestion
      catch (RootException const & e)
```